src/Workspaces/MSBuildTest/Utilities/DotNetSdkLocator.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text.Json.Nodes;
+
+namespace Microsoft.CodeAnalysis.MSBuild.UnitTests;
+
+internal class DotNetSdkLocator
+{
+    public static string? SdkPath { get; private set; } = null;
+
+    static DotNetSdkLocator()
+    {
+        if (TryGetGlobalJsonPath(out var globalJsonPath)
```

Comment:
So right now we assume dotnet.exe is on the path if you're going to be analyzing .NET Core projects:

https://github.com/dotnet/roslyn/blob/291d5d70fb062d1c8caf324b9369c535befaf6ed/src/Workspaces/Core/MSBuild/MSBuild/BuildHostProcessManager.cs#L154-L171

I'm not saying that's a _good_ thing (indeed we may have problems there) but if MSBuildWorkspace is assuming it, we can probably just assume it in tests and remove this file? This was always suspect anyways since if we run tests on different machines and such it's less clear if the global.json will end up getting copied around or not.

Reply:
In the Helix case we do rely on it using the version on Path as we do not locate the global.json. My only thought is that we really want to use the pinned version here, otherwise we may get a different set of templates. If that isn't a concern, deleting code is easy. =)

Reply:
And there's a potential problem here of course which is we're implicitly depending on whatever SDK version is installed, right? Because our global.json might not get considered?

Reply:
Now I think I grok what you were originally saying. You are saying, "yes, the template from the correct SDK will be used to generate the project, but a different SDK may be used by the Build Host".

In the Helix case Roslyn ensures the only installed dotnet matches the sdk pinned version.
https://github.com/dotnet/roslyn/blob/291d5d70fb062d1c8caf324b9369c535befaf6ed/src/Tools/Source/RunTests/TestRunner.cs#L100-L113

But, I think there is some concern here for the local developer experience. I will think on this.